### DISCUSSION TITLE
how does one apply sample() to genotype_matrix ?

### DISCUSSION THREAD

**Discussion by mshpak76** [tskit-dev/pyslim#354] [General]
In a previous version of my code, I created a data object using simplify() that I then used to sample from genotype_matrix, i.e.

    mts_subset_a = mts.simplify(mts.samples(population=1), filter_sites=False)
    Genotypes_A = mts_subset_a.genotype_matrix()

I would like to remove the simplify() function because of artifacts that it generates and sample directly from genotype_matrix, but just using:

    mts_subset_a = mts.samples(population=1)
    Genotypes_A = mts_subset_a.genotype_matrix()

throws an error

     AttributeError: 'numpy.ndarray' object has no attribute 'genotype_matrix'

so presumably mts.samples(population=1) did not create an object that could be applied to genotype_matrix. What is the correct syntax for this?

**Comment by benjeffery** [comment]
`ts.genotype_matrix` has a `samples` argument: https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence.genotype_matrix

**Reply by petrelharp** [reply]
This seems like a good answer - if this works for you, do you mind selecting it as "the right answer", @mshpak76 ?

**Reply by mshpak76** [reply]
Yes, this does work, thank you both for the response.
